File: cs6011/Week5/Day21/HelloAndroid/app/src/main/java/com/example/helloandroid/MainActivity.java
package com.example.helloandroid;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private boolean firstClick = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void handleClick(View view) {
        Log.i("Wk:MainActivity", "button was pressed");
        if (firstClick) {
            TextView roomName = findViewById(R.id.helloWorldTxt);
            roomName.setText("Enter a room name");
            firstClick = false;
        }else{
            //switch to 2nd activity
            Intent intent = new Intent(this, AnotherActivity.class);
            startActivity(intent);
        }

    }
}

File: cs6018/HelloAndroidStudio/app/src/main/java/com/example/helloandroidstudio/MainActivity.kt
package com.example.helloandroidstudio

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }
    fun onClick(v: View){
        val clickedButton = v as Button
        val selectedStudent = clickedButton.text.toString()
        Log.d("student", "the selected student is..... $selectedStudent")

        val studentIntent = Intent(this, MainActivity2::class.java)
        val studentBundle = Bundle()
        studentBundle.putString("student", selectedStudent)
        studentIntent.putExtras(studentBundle)
        startActivity(studentIntent)
    }
}